src/commands/deploy/deploy.ts

```diff
@@ -47,7 +46,9 @@ export async function deploy(actionContext: IActionContext, arg1?: vscode.Uri |
 
     // because this is workspace dependant, do it before user selects app
     await setPreDeployConfig(context);
-    const node: SiteTreeItem = await getDeployNode(context, ext.rgApi.tree, arg1, arg2, [ResolvedWebAppResource.webAppContextValue]);
+    const node: SiteTreeItem = await getDeployNode(context, ext.rgApi.tree, arg1, arg2, () => ext.rgApi.pickAppResource(context, {
```

Comment:
Just to double check, the `ResolvedWebAppResource.webAppContextValue` is no longer needed?